velox/type/Type.h

```diff
@@ -628,7 +678,9 @@ class DecimalType : public ScalarType<KIND> {
       KIND == TypeKind::SHORT_DECIMAL ? 18 : 38;
 
   DecimalType(const uint8_t precision = 18, const uint8_t scale = 0)
-      : precision_(precision), scale_(scale) {
+      : precision_(precision),
+        scale_(scale),
+        parameters_{TypeParameter::of(precision), TypeParameter::of(scale)} {
```

Comment:
Why do we need both (`precision_, scale_`) and `TypeParameters`? Can't we remove precision_, scale_?